src/image.rs

```diff
@@ -83,71 +83,63 @@ pub fn set_data(mut cx: FunctionContext) -> JsResult<JsBoolean> {
   let buffer = cx.argument::<JsBuffer>(1)?;
   let data = Data::new_copy(buffer.as_slice(&cx));
 
-  this.image = SkImage::from_encoded(data);
-  Ok(cx.boolean(this.image.is_some()))
-}
-
-pub fn load_svg(mut cx: FunctionContext) -> JsResult<JsBoolean> {
-  let this = cx.argument::<BoxedImage>(0)?;
-  let mut this = this.borrow_mut();
-
-  let buffer = cx.argument::<JsBuffer>(1)?;
-  let data = Data::new_copy(buffer.as_slice(&mut cx));
-
-  // Parse & load the SVG data.
-  let dom = svg::Dom::from_bytes(&data, FontMgr::default());
-  if !dom.is_ok() {
-    return cx.throw_type_error("Error loading SVG data.")
-  }
-  let mut dom = dom.unwrap();
-
-  // Get the intrinsic size of the `svg` root element as specified in the width/height attributes, if any.
-  // So far skia-safe doesn't provide direct access to the needed methods, so we have to go direct to the source.
-  let i_size = unsafe { *dom.inner().containerSize() };  // skia_bindings::SkSize
-  // let i_size = dom.inner().fContainerSize;  // "safe" but this is using a private member of the C++ class (somehow... skia-"safe" :-P )
-  // TODO: Switch to these once available in skia-safe 0.79+
-  // let mut root = dom.root();
-  // let i_size = root.intrinsic_size();
-
-  let mut bounds = Rect::from_wh(i_size.fWidth, i_size.fHeight);
-
-  // Set a flag to indicate that the image doesn't have its own intrinsic size.
-  // This may be used at drawing time if user doesn't specify a size in `drawImage()`,
-  // in which case the the canvas' size will be used as the image size.
-  // This is a "complication" to match Chrome's behavior... one could argue that it should
-  // just be drawn at the default size (set below). Which is what FF does (though that has its own anomalies).
-  this.adjust_size_to_canvas = bounds.is_empty();
-
-  // Check if width/height are valid attribute values in the root `<svg>` element.
-  // If w/h aren't specified in an SVG (which is not uncommon), both Chrome and FF will:
-  //  - If only one dimension is missing then use the same size for both;
-  //  - If both are missing then assign a default of 150 (which seems arbitrary but I guess as good as any);
-  // `Dom::containerSize()` will return zero for both width and height if _either_ attribute is missing from `<svg>`.
-  // This seems a bit suspicious (as in may change in future?), so in the interest of paranoia let's check them individually.
-  // TODO: See if we can get actual width/height attribute values from DOM with skia-safe 0.79+
-  if bounds.right <= 0.0 {
-    bounds.right = match bounds.bottom > 0.0 {
-      true  => bounds.bottom,
-      false => 150.0f32
-    };
-  }
-  if bounds.bottom <= 0.0 {
-    bounds.bottom = bounds.right;
-  }
+  this.image = SkImage::from_encoded(&data);
+  if this.image.is_none(){
```

Comment:
So did you test this?  I'm extremely skeptical and wouldn't do it this way. Just to save some JS code?  I don't get it. Besides performance issues, throwing random bytes at the Skia image decoder seems like a path to doomstown.

Reply:
Yup. Here's what passing svg data to `Image::from_encoded` eats up:
<img width="547" alt="image" src="https://github.com/user-attachments/assets/a0535ce7-ba7c-4f67-847c-88cabe625419">
<img width="92" alt="image" src="https://github.com/user-attachments/assets/e1a376e4-f19d-45dd-9e2e-b2ba52010efa">

And if I set `iterations` to 1 to see if it was all caching or something:
<img width="117" alt="image" src="https://github.com/user-attachments/assets/47011e16-eff8-4752-ab36-0f977057ec32">

Reply:
So it is caching? But yes 250us is impressive, if not quite free.
The "contract" doesn't say it won't crash or overflow some buffer though... lol  Me, paranoid?  Naaaah....

Sort of BTW, I'm trying out [images::deferred_from_encoded_data](https://rust-skia.github.io/doc/skia_safe/image/images/fn.deferred_from_encoded_data.html) instead which seems to work nicely.  Kinda goes with the pre-rendering vs. "do it all at the end" strategy choice, if I understand it correctly.

Reply:
Agreed! The deferred option really does sound like a good match for how the rendering timeline works right now.

Reply:
Also, do we know what the SVG parser is going to do with data that is an invalid bitmap (for whatever reason), but isn't an SVG?

Reply:
When you pass it something non-SVG (e.g., setting `src`=`lib/v6/index.node`) it throws:
```
TypeError: Error loading SVG data.
```

Reply:
Okay this is better: 82e2be8a7b41b8b01ed97098948384c6849ac6e8